src/interactWithTerminal.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const readline = require('readline');
+
+const { generateNumber } = require('./generateNumber');
+const { calculateBullsAndCows } = require('./calculateBullsAndCows');
+
+const terminal = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const interactWithTerminal = () => {
+  terminal.question('Try to guess: ', (guess) => {
+    for (const char of guess) {
+      if ((guess.indexOf(char) !== guess.lastIndexOf(char))
+      || isNaN(+char)
+      || guess.length !== 4) {
+        throw new Error('Incorrect input');
```

Comment:
Would be better to add some text about why it is incorrect input or which input will be correct